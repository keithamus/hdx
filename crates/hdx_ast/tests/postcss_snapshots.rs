mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
