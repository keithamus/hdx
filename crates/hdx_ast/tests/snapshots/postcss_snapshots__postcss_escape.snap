---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Html": {
                  "A": {
                    "kind": "Ident",
                    "offset": 0,
                    "len": 1
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 2,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 8,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 18,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 19,
                "len": 1
              },
              {
                "type": "Function",
                "name": {
                  "kind": "Function",
                  "offset": 20,
                  "len": 4
                },
                "values": [
                  {
                    "kind": "String",
                    "offset": 24,
                    "len": 8
                  }
                ],
                "close_paren": {
                  "kind": "RightParen",
                  "offset": 32,
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 33,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 39,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 49,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 50,
                "len": 1
              },
              {
                "kind": "Url",
                "offset": 51,
                "len": 8
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 59,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 65,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 75,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 76,
                "len": 1
              },
              {
                "kind": "Url",
                "offset": 77,
                "len": 8
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 85,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 91,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 101,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 102,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 103,
                "len": 3
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 106,
            "len": 1
          }
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 108,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "functional-pseudo-class",
              "value": {
                "type": "not",
                "value": {
                  "colon": {
                    "kind": "Colon",
                    "offset": 111,
                    "len": 1
                  },
                  "function": {
                    "kind": "Function",
                    "offset": 112,
                    "len": 4
                  },
                  "value": [
                    {
                      "components": [
                        {
                          "type": "attribute",
                          "value": {
                            "type": "Attribute",
                            "open": {
                              "kind": "LeftSquare",
                              "offset": 116,
                              "len": 1
                            },
                            "namespace_prefix": null,
                            "attribute": {
                              "kind": "Ident",
                              "offset": 117,
                              "len": 3
                            },
                            "operator": {
                              "type": "Exact",
                              "value": {
                                "kind": "Delim",
                                "offset": 120,
                                "len": 1
                              }
                            },
                            "value": {
                              "type": "String",
                              "value": {
                                "kind": "String",
                                "offset": 121,
                                "len": 3
                              }
                            },
                            "modifier": null,
                            "close": {
                              "kind": "RightSquare",
                              "offset": 124,
                              "len": 1
                            }
                          }
                        }
                      ],
                      "comma": null
                    }
                  ],
                  "close": {
                    "kind": "RightParen",
                    "offset": 125,
                    "len": 1
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 127,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 129,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "functional-pseudo-class",
              "value": {
                "type": "not",
                "value": {
                  "colon": {
                    "kind": "Colon",
                    "offset": 131,
                    "len": 1
                  },
                  "function": {
                    "kind": "Function",
                    "offset": 132,
                    "len": 4
                  },
                  "value": [
                    {
                      "components": [
                        {
                          "type": "tag",
                          "value": {
                            "Html": {
                              "Div": {
                                "kind": "Ident",
                                "offset": 136,
                                "len": 3
                              }
                            }
                          }
                        }
                      ],
                      "comma": null
                    }
                  ],
                  "close": {
                    "kind": "RightParen",
                    "offset": 144,
                    "len": 1
                  }
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 146,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 148,
        "len": 1
      }
    },
    {
      "type": "QualifiedRule",
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 149,
            "len": 1
          },
          {
            "kind": "Colon",
            "offset": 150,
            "len": 1
          },
          {
            "type": "Function",
            "name": {
              "kind": "Function",
              "offset": 151,
              "len": 4
            },
            "values": [
              {
                "kind": "Colon",
                "offset": 155,
                "len": 1
              },
              {
                "type": "Function",
                "name": {
                  "kind": "Function",
                  "offset": 156,
                  "len": 10
                },
                "values": [
                  {
                    "kind": "Dimension",
                    "offset": 166,
                    "len": 2,
                    "unit": "unknown"
                  },
                  {
                    "kind": "Whitespace",
                    "offset": 168,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 169,
                    "len": 2
                  },
                  {
                    "kind": "Whitespace",
                    "offset": 171,
                    "len": 1
                  },
                  {
                    "type": "SimpleBlock",
                    "start": 172,
                    "open": {
                      "kind": "LeftSquare",
                      "len": 1
                    },
                    "values": {
                      "values": [
                        {
                          "kind": "Ident",
                          "offset": 173,
                          "len": 3
                        },
                        {
                          "kind": "Delim",
                          "offset": 176,
                          "len": 1
                        },
                        {
                          "kind": "String",
                          "offset": 177,
                          "len": 3
                        }
                      ]
                    },
                    "close": {
                      "kind": "RightSquare",
                      "len": 1
                    }
                  }
                ],
                "close_paren": {
                  "kind": "RightParen",
                  "offset": 181,
                  "len": 1
                }
              }
            ],
            "close_paren": {
              "kind": "RightParen",
              "offset": 182,
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 183,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 184,
          "len": 1
        },
        "declarations": [],
        "rules": [],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 186,
          "len": 1
        }
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "attribute",
              "value": {
                "type": "Attribute",
                "open": {
                  "kind": "LeftSquare",
                  "offset": 188,
                  "len": 1
                },
                "namespace_prefix": null,
                "attribute": {
                  "kind": "Ident",
                  "offset": 189,
                  "len": 3
                },
                "operator": {
                  "type": "Exact",
                  "value": {
                    "kind": "Delim",
                    "offset": 192,
                    "len": 1
                  }
                },
                "value": {
                  "type": "Ident",
                  "value": {
                    "kind": "Ident",
                    "offset": 193,
                    "len": 2
                  }
                },
                "modifier": null,
                "close": {
                  "kind": "RightSquare",
                  "offset": 195,
                  "len": 1
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 197,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 199,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "attribute",
              "value": {
                "type": "Attribute",
                "open": {
                  "kind": "LeftSquare",
                  "offset": 201,
                  "len": 1
                },
                "namespace_prefix": null,
                "attribute": {
                  "kind": "Ident",
                  "offset": 202,
                  "len": 3
                },
                "operator": {
                  "type": "Exact",
                  "value": {
                    "kind": "Delim",
                    "offset": 205,
                    "len": 1
                  }
                },
                "value": {
                  "type": "Ident",
                  "value": {
                    "kind": "Ident",
                    "offset": 206,
                    "len": 2
                  }
                },
                "modifier": null,
                "close": {
                  "kind": "RightSquare",
                  "offset": 208,
                  "len": 1
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 210,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 212,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "attribute",
              "value": {
                "type": "Attribute",
                "open": {
                  "kind": "LeftSquare",
                  "offset": 214,
                  "len": 1
                },
                "namespace_prefix": null,
                "attribute": {
                  "kind": "Ident",
                  "offset": 215,
                  "len": 3
                },
                "operator": {
                  "type": "Exact",
                  "value": {
                    "kind": "Delim",
                    "offset": 218,
                    "len": 1
                  }
                },
                "value": {
                  "type": "Ident",
                  "value": {
                    "kind": "Ident",
                    "offset": 219,
                    "len": 2
                  }
                },
                "modifier": null,
                "close": {
                  "kind": "RightSquare",
                  "offset": 221,
                  "len": 1
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 223,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 225,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "attribute",
              "value": {
                "type": "Attribute",
                "open": {
                  "kind": "LeftSquare",
                  "offset": 227,
                  "len": 1
                },
                "namespace_prefix": null,
                "attribute": {
                  "kind": "Ident",
                  "offset": 228,
                  "len": 3
                },
                "operator": {
                  "type": "Exact",
                  "value": {
                    "kind": "Delim",
                    "offset": 231,
                    "len": 1
                  }
                },
                "value": {
                  "type": "Ident",
                  "value": {
                    "kind": "Ident",
                    "offset": 232,
                    "len": 20
                  }
                },
                "modifier": null,
                "close": {
                  "kind": "RightSquare",
                  "offset": 252,
                  "len": 1
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 254,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 256,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Unknown": {
                  "kind": "Ident",
                  "offset": 259,
                  "len": 6
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 266,
        "len": 1
      },
      "declarations": [],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 268,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "tag",
              "value": {
                "Unknown": {
                  "kind": "Ident",
                  "offset": 271,
                  "len": 4
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 275,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 281,
            "len": 5
          },
          "colon": {
            "kind": "Colon",
            "offset": 286,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 287,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 288,
                "len": 2
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 290,
            "len": 1
          }
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 292,
        "len": 1
      }
    },
    {
      "type": "QualifiedRule",
      "prelude": {
        "values": [
          {
            "kind": "Whitespace",
            "offset": 293,
            "len": 2
          },
          {
            "type": "SimpleBlock",
            "start": 295,
            "open": {
              "kind": "LeftSquare",
              "len": 1
            },
            "values": {
              "values": [
                {
                  "kind": "Ident",
                  "offset": 296,
                  "len": 4
                },
                {
                  "kind": "Delim",
                  "offset": 300,
                  "len": 1
                },
                {
                  "kind": "Semicolon",
                  "offset": 301,
                  "len": 1
                }
              ]
            },
            "close": {
              "kind": "RightSquare",
              "len": 1
            }
          }
        ]
      },
      "block": {
        "type": "Block",
        "start": 303,
        "open_curly": {
          "kind": "LeftCurly",
          "offset": 304,
          "len": 1
        },
        "declarations": [],
        "rules": [],
        "close_curly": {
          "kind": "RightCurly",
          "offset": 306,
          "len": 1
        }
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "class",
              "value": {
                "type": "Class",
                "dot": {
                  "kind": "Delim",
                  "offset": 309,
                  "len": 1
                },
                "name": {
                  "kind": "Ident",
                  "offset": 310,
                  "len": 4
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 315,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 321,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 329,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 330,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 331,
                "len": 3
              }
            ]
          },
          "important": null,
          "semicolon": null
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 335,
        "len": 1
      }
    }
  ]
}
